lua/smear_cursor/animation.lua

```diff
@@ -231,7 +235,7 @@ local function clamp_to_buffer(position)
 end
 
 local function scroll_buffer_space()
-	if current_top_row ~= previous_top_row then
+	if current_top_row ~= previous_top_row and current_line ~= previous_line then
```

Comment:
I don't understand the change though 🤔

Why tracking for the line nr in the buffer and doing something when it's different has any behavior change when the line nr don't actually change with `z*` actions?

Reply:
The `scroll_buffer_space` function shifts the smear's origin (in screen space) before drawing it, to take the buffer's scroll into account. This is only needed when doing `<c-u>` or `<c-d>` (`current_top_row ~= previous_top_row` and `current_line ~= previous_line`).

This PR prevents the shift for `z*` movements (`current_top_row ~= previous_top_row` and **`current_line == previous_line`**).